src/rpk_entity.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015-2021 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use crate::error::Error;
+use crate::{der, signed_data, DerTypeId};
+use pki_types::{SignatureVerificationAlgorithm, SubjectPublicKeyInfoDer};
+use signed_data::SubjectPublicKeyInfo;
+
+/// A Raw Public Key certificate, used for connections using raw public keys as specified in RFC7250.
+#[derive(Debug)]
+pub struct RawPublicKeyEntity<'a> {
+    inner: SubjectPublicKeyInfo<'a>,
```

Comment:
I suggest a different approach to this. At the moment, we are completely dissecting the SPKI, then building it back up to use it.

Instead, how about this just contains an `untrusted::Input<'a>` that can be used directly by `signed_data::verify_signature`.

The `try_from` impl should do everything it currently does, but output `untagged_spki` rather than `spki`.
 
That should allow eliminating the new `SubjectPublicKeyInfo::subject_public_key_info` function, and save the allocations in the signature verification path.

Reply:
Thanks, good idea! That makes everything a lot less complex